src/nnet3/nnet-simple-component.h

```diff
@@ -309,6 +310,7 @@ class AffineComponent: public UpdatableComponent {
   Component *CollapseWithNext(const FixedAffineComponent &next) const;
```

Comment:
@hhadian, could you please also modify nnet-component-test.cc to add a test
for the Vectorize and Unvectorize() ?

This could be done using similar code to the code in
TestNnetComponentAddScale.
This could also test SetZero().
You could copy a component using Copy() and then set it to zero (use
treat_as_gradient == false), and then if ParameterDim() != 0, test that the
vectorized parameters are in fact zero.  Then vectorize the parameters of
the original component, call UnVectorize on the new component using the
resulting parameter vector, and make sure that the new component's
parameters are the same as the old ones (e.g. by comparing the info strings
and checking that the DotProduct calls return the same value).

You should also test that if the component does not say it is updatable
(kUpdatable == false), then the ParameterDim() is zero, and that Add(),
Scale(), and SetZero() are all no-ops, and that DotProduct returns zero...
again, you can use the you could also use the info-string test to check
this.  We are testing that non-updatable components behave as if they don't
have any parameters from the point of view of those functions:  this is
necessary to check that components that are 'optionally updatable', which
they may be in future, have consistent behavior if they are set as not
updatable.

Dan


On Mon, Nov 2, 2015 at 5:12 PM, Vijayaditya Peddinti <
notifications@github.com> wrote:

> In src/nnet3/nnet-simple-component.h
> <https://github.com/kaldi-asr/kaldi/pull/307#discussion_r43690717>:
>
> > @@ -309,6 +310,7 @@ class AffineComponent: public UpdatableComponent {
> >    Component *CollapseWithNext(const FixedAffineComponent &next) const;
>
> @danpovey <https://github.com/danpovey>
> A clean-up of the unused code from Affine and other components, to ensure
> that the unused methods are not copied into newer classes, might be
> necessary.
>
> Also there is no test for the Vectorize/Unvectorize methods. The
> Convolution*Components don't implement these methods and there was never a
> failed test.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/307/files#r43690717>.
>